### DISCUSSION TITLE
NetworkUtil module function call

### DISCUSSION THREAD

**Discussion by ytrewq271828** [ANLAB-KAIST/KENSv3#167] [Project Q&A 2. Transport Layer (TCP)]
I've encountered an error while calling functions defined in NetworkUtil module.
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/52207623/3b986f45-8172-4199-a8f9-aec9ce1a7b55)

I think the type of the argument matches the given implementation, but my IDE keeps returning the following error message.

_no instance of function template "E::NetworkUtil::arrayToUINT64" matches the argument listC/C++(304)
TCPAssignment.cpp(335, 22): argument types are: (std::optional<E::ipv4_t>)_

Can you tell me why this error happens?
Thanks in advance.

**Comment by Pusnow** [comment]
Your `dstAddr`'s type seems `std::optional<E::ipv4_t>`.

To convert `std::optional<E::ipv4_t>` to `E::ipv4_t`, you should follow the usage of `std::optional`.

https://en.cppreference.com/w/cpp/utility/optional/value
